clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -223,6 +223,8 @@ public void add(CompoundStat stat) {
      *        sensor.
      */
     public synchronized void add(CompoundStat stat, MetricConfig config) {
+        if (hasExpired())
```

Comment:
nit: maybe we may as well make a private method, say `ensureNotExpired` or something like that.

Reply:
Updated the PR.